### ISSUE TITLE
[HOLD for payment 2024-07-26] [$250] Workspace - "Delay submissions" turns off if the "Submission frequency" is set to "Manually"

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#43196]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.4.80-1
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught during regression testing, add the test name, ID and link from TestRail:** N/A
**Issue reported by:** Applause - Internal Team

## Action Performed:
1. Log in with a new expensifail account
2. Create a workspace
3. Turn "Workflows" on
4. Navigate to Workflows
5. Turn "Delay submissions" on
6. Set "Submission frequency" to "Manually"

## Expected Result:
"Delay submissions" should remain on

## Actual Result:
"Delay submissions" turns off if the "Submission frequency" is set to "Manually"

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

</details> 

https://github.com/Expensify/App/assets/78819774/a9771da7-3b1b-4576-b9fb-acc208f2f8c2

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~015bf509d8291c1690</li>
        <li>Upwork Job ID: 1800519449372778818</li>
        <li>Last Price Increase: 2024-06-25</li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @dukenv0307</details>

**Comment by lanitochka17** [comment]
@sonialiap  FYI I haven't added the External label as I wasn't 100% sure about this issue. Please take a look and add the label if you agree it's a bug and can be handled by external contributors

**Comment by Tony-MK** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.

Workspace - "Delay submissions" turns of if the "Submission frequency" is set to "Manually"

### What is the root cause of that problem?

The root cause of the problem is that when a user sets it to manual, the `policy?.harvesting?.enabled` becomes false.

The `policy?.harvesting?.enabled` is false because it checks whether the scheduled submission is enabled which won't be in this situation.

https://github.com/Expensify/App/blob/f3bb5a5a29aa752440190e961d1386d26bee5cf8/src/types/onyx/Policy.ts#L777-L781

Hence, the condition that makes the `isActive` in the `WorkspaceWorkflowsPage` becomes `false` when the auto-reporting frequency is manual.

https://github.com/Expensify/App/blob/f3bb5a5a29aa752440190e961d1386d26bee5cf8/src/pages/workspace/workflows/WorkspaceWorkflowsPage.tsx#L132

### What changes do you think we should make in order to solve the problem?


Therefore, we must consider the `autoReportingFrequency` if it is `CONST.POLICY.AUTO_REPORTING_FREQUENCIES.MANUAL`.

Just like we did with the `CONST.POLICY.AUTO_REPORTING_FREQUENCIES.INSTANT`.

So, let's change the `isActive` logic to something simpler than the one below.

```
isActive: (((policy?.harvesting?.enabled && policy.autoReportingFrequency !== CONST.POLICY.AUTO_REPORTING_FREQUENCIES.INSTANT ) ?? (policy?.harvesting?.enabled && policy.autoReportingFrequency !== CONST.POLICY.AUTO_REPORTING_FREQUENCIES.MANUAL) ) && !hasDelayedSubmissionError) ?? false,
```

**Comment by sonialiap** [comment]
@abdulrahuman5196 what do you think of the above proposal?

**Comment by abdulrahuman5196** [comment]
checking now

**Comment by abdulrahuman5196** [comment]
Hi all, I dont see the Delay submissions option in the workflow? Do I need to be part of some beta or any specific type of workspace?

<img width="1512" alt="Screenshot 2024-06-20 at 9 28 06 PM" src="https://github.com/Expensify/App/assets/46707890/82656c6d-a0de-4b53-a30d-78d02e14a2fc">

**Comment by abdulrahuman5196** [comment]
Hi @sonialiap  Sorry for the delay, I have limited availability at the moment. So unassigning myself. Kindly reassign another C+ for review.

**Comment by sonialiap** [comment]
@dukenv0307 added you as the C+

**Comment by sonialiap** [comment]
@dukenv0307 
>After choosing Manual option, harvesting.enable is false and frequency is immediate. Do you think it's BE bug? If not, why don't we just remove the manual option, since we can toggle the delay submission instead of choosing Manual?

What do you mean by "toggle the delay submission"? Do you mean disable it? The point of Manual is to allow expenses to pile up before being submitted, like Weekly or Monthly but without an end date. If Delay Submission is disabled (I'm assuming that's what you mean by toggling it) then expenses are submitted immediately, so there's no delay as the customer might want. I think we need to keep the Manual option and not disable Delay Submission.

**Comment by sonialiap** [comment]
I don't know how the functionality looks like in the code but behavior wise, when set to Manual I expect it to be
- harvesting enabled
- frequency (is this submission frequency? If submission then disabled, we don't want automatic submission)

@roryabraham could you confirm english to code for me haha? We want this feature to function like OD Scheduled Submit where when "Manual" is selected expenses are harvested onto reports but are not automatically submitted

**Comment by roryabraham** [comment]
Ok, I've looked into this and actually @Tony-MK was pretty close. This logic is actually pretty convoluted, but I'll try my best to spell it out. (for what it's worth, I can't imagine it would've been possible to figure all this out without access to the back-end code)

- There's a policy field called `autoReportingFrequency`, which can have the following values:
    - `immediate`
    - `instant`
    - `weekly`
    - `semimonthly`
    - `monthly`
    - `trip`
    - `manual`
    - `lastDayOfMonth`
    - `lastBusinessDayOfMonth`
- There's another policy field called `harvesting.enabled`, that's a boolean that determines whether harvesting runs
- Harvesting is the process by which a daily job runs for each policy and determines if any reports need to be submitted. It also handles monthly summary emails and other stuff. ([internal docs](https://github.com/Expensify/Web-Expensify/blob/master/_devportal/docs/harvesting.md))
- confusingly, `harvesting.enabled` doesn't actually prevent the harvesting job from running (because of the monthly summary email I mentioned). It just prevents reports from being submitted as part of harvesting.
- `instant` = "instant submit", was added recently as part of the [Instant Submit for Collect](https://docs.google.com/document/d/1HIPlPKDyJEX73bUGceo354tvU7xqXkhj9j14GoLhMJw/edit) doc. As the name suggests, it means that when a report is created in the policy it's instantly submitted.
- `immediate` is a legacy term. It is in fact, not immediate 😂 
- `immediate` & `harvesting.enabled` = `daily`
- `immediate` & `!harvesting.enabled` = `manual`

ok, so with that out of the way, I _think_ we can fix this issue a bit more simply by ignoring `harvesting.enabled`

```diff
diff --git a/src/pages/workspace/workflows/WorkspaceWorkflowsPage.tsx b/src/pages/workspace/workflows/WorkspaceWorkflowsPage.tsx
index 372044d4bb..ae5cc5edeb 100644
--- a/src/pages/workspace/workflows/WorkspaceWorkflowsPage.tsx
+++ b/src/pages/workspace/workflows/WorkspaceWorkflowsPage.tsx
@@ -128,7 +128,7 @@ function WorkspaceWorkflowsPage({policy, betas, route}: WorkspaceWorkflowsPagePr
                                   brickRoadIndicator={hasDelayedSubmissionError ? CONST.BRICK_ROAD_INDICATOR_STATUS.ERROR : undefined}
                               />
                           ),
-                          isActive: (policy?.harvesting?.enabled && policy.autoReportingFrequency !== CONST.POLICY.AUTO_REPORTING_FREQUENCIES.INSTANT && !hasDelayedSubmissionError) ?? false,
+                          isActive: (policy?.autoReportingFrequency !== CONST.POLICY.AUTO_REPORTING_FREQUENCIES.INSTANT && !hasDelayedSubmissionError) ?? false,
                           pendingAction: policy?.pendingFields?.autoReporting,
                           errors: ErrorUtils.getLatestErrorField(policy ?? {}, CONST.POLICY.COLLECTION_KEYS.AUTOREPORTING),
                           onCloseError: () => Policy.clearPolicyErrorField(policy?.id ?? '-1', CONST.POLICY.COLLECTION_KEYS.AUTOREPORTING),
```

Going through the options:

- `immediate` & `harvesting.enabled` = `daily` => so `Delay Submissions` should be active
- `immediate` & `!havesting.enabled` = `manual` => so `Delay Submissions` should be active
- Really the only time when we wouldn't want `Delay Submissions` to be active is if the policy has the new `instant` auto-reporting frequency (or apparently if there's some error, but I don't have good context about that)

**Comment by roryabraham** [comment]
Will be fixing this myself in https://github.com/Expensify/App/pull/45540

**Comment by roryabraham** [comment]
@dukenv0307 PR is ready for your review: https://github.com/Expensify/App/pull/45540

**Comment by roryabraham** [comment]
PR is C+ approved, just waiting on review from @Beamanator, @mountiny, and/or @madmax330

**Comment by Tony-MK** [comment]
Hey, @roryabraham, I am curious to know if I am eligible for compensation for providing a solution that was used in the recently merged [PR](https://github.com/Expensify/App/pull/45540) based on your previous [comment](https://github.com/Expensify/App/issues/43196#issuecomment-2232030963).

Thank you

**Comment by sonialiap** [comment]
@roryabraham what do you think? https://github.com/Expensify/App/issues/43196#issuecomment-2246975341

**Comment by sonialiap** [comment]
Payment summary:
- @dukenv0307 $250 - offer sent in upwork - please complete the [checklist](https://github.com/Expensify/App/issues/43196#issuecomment-2239979912)

https://www.upwork.com/ab/applicants/1816797650551015987/job-details

**Comment by roryabraham** [comment]
> Hey, @roryabraham, I am curious to know if I am eligible for compensation for providing a solution that was used in the recently merged

Sorry, but no I don't think so. I figured out the solution by looking at back-end code, and the proposal you posted wasn't a complete and correct solution. I merely meant to acknowledge that out of all the proposals yours was the closest to being correct.

**Comment by sonialiap** [comment]
@dukenv0307 please accept the offer in upwork 💰

**Comment by sonialiap** [comment]
@dukenv0307 it seems like I sent the offer to that profile, I just sent a message on the offer. Let me know if you don't see it then I'll resend :D

**Comment by sonialiap** [comment]
Oh I misnamed the offer with another issue number 🤦 I called it `44830`

**Comment by sonialiap** [comment]
Paid ✔️
